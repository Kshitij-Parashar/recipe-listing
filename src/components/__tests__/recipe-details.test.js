import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import RecipeDetailsContainer from '../recipe-details/recipe-details-container';
import '../../setupTests';

it('should render correctly <RecipeDetailsContainer>', () => {
  const historyMock = {
    goBack: jest.fn(),
    listen: jest.fn(),
    location: { pathname: '' },
  };
  const mockStore = configureStore();

  const store = mockStore({
    recipe: {
      meals: [
        {
          idMeal: '52874',
          strMeal: 'Beef and Mustard Pie',
          strMealThumb:
            'https://www.themealdb.com/images/media/meals/sytuqu1511553755.jpg',
        },
      ],
      loading: false,
      error: false,
    },
  });
  const wrapper = mount(
    <Provider store={store}>
      <Router history={historyMock}>
        <RecipeDetailsContainer />
      </Router>
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
