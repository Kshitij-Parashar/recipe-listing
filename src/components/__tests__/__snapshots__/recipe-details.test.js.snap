// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly <RecipeDetailsContainer> 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "goBack": [MockFunction],
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "location": Object {
          "pathname": "",
        },
      }
    }
  >
    <Connect(RecipeDetailsContainer)>
      <RecipeDetailsContainer
        error={false}
        loadRecipe={[Function]}
        loading={false}
      >
        <div
          className="recipe-details"
        >
          <RecipeDetail
            error={false}
            goBack={[Function]}
            loadRecipe={[Function]}
            loading={false}
            parsedRecipe={null}
          >
            <h2>
              Recipe not found.
            </h2>
          </RecipeDetail>
        </div>
      </RecipeDetailsContainer>
    </Connect(RecipeDetailsContainer)>
  </Router>
</Provider>
`;
